proc/community/src/main/java/org/neo4j/gds/leiden/LeidenStreamSpec.java

```diff
@@ -60,19 +64,37 @@ public ComputationResultConsumer<Leiden, LeidenResult, LeidenStreamConfig, Strea
             var graph = computationResult.graph();
             boolean includeIntermediateCommunities = computationResult.config().includeIntermediateCommunities();
 
-            boolean consecutiveIds = computationResult.config().consecutiveIds();
-            var nodeProperties = consecutiveIds ? nodeProperties(computationResult) : null;
-            var communities = leidenResult.communities();
 
             return LongStream.range(0, graph.nodeCount())
-                .mapToObj(nodeId -> new StreamResult(
-                    graph.toOriginalNodeId(nodeId),
-                    includeIntermediateCommunities ? leidenResult.getIntermediateCommunities(nodeId) : null,
-                    consecutiveIds ? nodeProperties.longValue(nodeId) : communities.get(nodeId)
-                ));
+                .mapToObj(nodeId -> {
+                    Optional<Long> communityId = getCommunityId(computationResult, nodeId);
+
+                    return communityId.map(id -> new StreamResult(
+                            graph.toOriginalNodeId(nodeId),
+                            includeIntermediateCommunities ? leidenResult.getIntermediateCommunities(nodeId) : null,
+                            id
+                    )).orElse(null);
+                })
+                .filter(Objects::nonNull);
         };
     }
 
+    @Nullable
+    private Optional<Long> getCommunityId(ComputationResult<Leiden, LeidenResult, LeidenStreamConfig> computationResult, long nodeId) {
+        var leidenResult = computationResult.result();
+        var communities = leidenResult.communities();
+        boolean consecutiveIds = computationResult.config().consecutiveIds();
+
+        if (!consecutiveIds) {
+            return Optional.of(communities.get(nodeId));
+        }
```

Comment:
we are running these lines for each node.
I would suggest to not pass the whole ComputationResult to this method but the pre-compute as much as possible outside (as done before).